src/sys/windows/selector.rs

```diff
@@ -371,14 +411,233 @@ impl Selector {
+                    .map(|r| r.into_event(self.token))
+            }
+        }
+    }
+
+    /// A `Binding` is embedded in all I/O objects associated with a `Registry`.
+    ///
+    /// Each registration keeps track of which selector the I/O object is
+    /// associated with, ensuring that implementations of `Source` can be
+    /// conformant for the various methods on Windows.
+    ///
+    /// If you're working with custom IOCP-enabled objects then you'll want to
+    /// ensure that one of these instances is stored in your object and used in the
+    /// implementation of `Source`.
+    ///
+    /// For more information about how to use this see the `windows` module
+    /// documentation in this crate.
+    #[derive(Debug, Clone)]
+    pub struct Binding {
+        selector: Arc<SelectorInner>,
+        token: Token
+    }
+
+    impl Binding {
+        /// Creates a new blank binding ready to be inserted into an I/O
+        /// object.
+        ///
+        /// Won't actually do anything until `register_handle` has been called during a call to
+        /// `Registry::register`.
+        pub fn new(registry: &Registry, token: Token) -> Self {
+            let selector = poll::selector(registry);
+            Binding { selector: Arc::clone(&selector.inner), token }
+        }
+
+        /// Register a raw windows handle with the IOCP event loop.
+        ///
+        /// The underlying resource needs to support overlapped operations. Thus, files and similar
+        /// entities need to be opened in overlapped mode.
+        ///
+        /// While it is safely possible to call this function more than once it is discouraged to do
+        /// so as the handler will use the same token again when generating events. This makes it hard
+        /// for the user of the registered `Source`.
+        pub fn register_handle<H>(&self, handle: &H) -> io::Result<()> where H: AsRawHandle + ?Sized {
+            let handler = RawHandleCompletionHandler {
+                token: self.token
+            };
+            self.selector.cp.register_handle(handle, handler.into())
+        }
+
+        /// Inject an event to the event loop.
+        ///
+        /// In case it is required to generate an event even though there was no callback due to a
+        /// completed IO operation, this method can be used to mark the associated token as ready, e.g.
+        /// in case it is necessary to signal an initial readiness before any IO operation has been
+        /// scheduled.
+        pub fn inject_event(&self, readiness: Readiness) {
+            let event = readiness.into_event(self.token);
+            self.selector.injected_events.lock().unwrap().push(event)
+        }
+
+        /// Check whether the provided registry is the same as the one used during creation of the
+        /// binding.
+        ///
+        /// IOCP driven sources cannot unregister from their completion port before they get closed.
```

Comment:
No `event::Source` in Mio can be registered with another `Poll`, so this is true for all platforms. See `IoSource`'s `selector_id` field.

Reply:
Ah, I see, so this is redundand and can be removed. I'll do that.